esmvaltool/diag_scripts/shared/_base.py

```diff
@@ -555,6 +566,20 @@ def main(cfg):
         logger.info("Removing %s from previous run.", provenance_file)
         os.remove(provenance_file)
 
-    yield cfg
+    if not args.no_distributed and 'scheduler_address' in cfg:
+        try:
+            client = distributed.Client(cfg['scheduler_address'])
+        except OSError:
+            logger.error(
+                "Unable to connect to the Dask distributed scheduler at %s. "
+                "If the scheduler is no longer available, try re-running the "
+                "diagnostic script with the --no-distributed flag.",
+                cfg['scheduler_address'])
+            raise
```

Comment:
I think it nicer to include the error message in an exception than calling `logger.error`, i.e., something like `raise Exception("...") from ...` where `Exception` should be something sensible. In my experience, users only look at the exceptions and tend to overlook error messages further up.

Reply:
Done in [cfd97aa](https://github.com/ESMValGroup/ESMValTool/pull/3151/commits/cfd97aaa3b2683531e508027882b371ddb5b2554)